src/org/mozartoz/truffle/nodes/builtins/BrowserBuiltins.java

```diff
@@ -7,18 +7,19 @@
 import com.oracle.truffle.api.dsl.NodeChild;
 import com.oracle.truffle.api.dsl.NodeChildren;
 import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.object.DynamicObject;
 
 public abstract class BrowserBuiltins {
 
+	@Builtin(tryDeref = 0)
```

Comment:
`tryDeref` means deref if it's bound but otherwise leave it untouched.
As a consequence, there should be a `@Specialization` below for the `Variable` type (our version of Transient),
and a negative guard for the other case.
Here it would work since DynamicObject are not Variable but I'd rather have this more explicit :smiley: